makefiles/setup_dirs.inc.mk

```diff
@@ -0,0 +1,77 @@
+# either in a Makefile, or as environment variable.
+ifeq ($(INSIDE_DOCKER),0)
+  # In Docker absolute paths are always given, so only fail when not in docker
+  ifeq ($(origin EXTERNAL_BOARD_DIRS),command line)
+    $(error EXTERNAL_BOARD_DIRS must be passed as environment variable, and not as command line argument)
+  endif
+  ifeq ($(origin EXTERNAL_MODULE_DIRS),command line)
+    $(error EXTERNAL_MODULE_DIRS must be passed as environment variable, and not as command line argument)
+  endif
+  ifeq ($(origin EXTERNAL_PKG_DIRS),command line)
+    $(error EXTERNAL_PKG_DIRS must be passed as environment variable, and not as command line argument)
+  endif
+endif
+
+# Deprecation of configuring 'RIOTBOARD'
+ifneq ($(abspath $(RIOTBASE)/boards),$(abspath $(RIOTBOARD)))
+  $(warning overriding RIOTBOARD for external boards is deprecated, use EXTERNAL_BOARD_DIRS instead)
+  override RIOTBOARD      := $(abspath $(RIOTBOARD))
+  __DIRECTORY_VARIABLES += RIOTBOARD
+endif
+
+# Only warn users, not the CI.
+ifneq ($(RIOT_CI_BUILD),1)
+  # Do not warn when set from sub-make
+  ifeq ($(MAKELEVEL),0)
+    ifneq (,$(BOARDSDIR))
+      $(warning Using BOARDSDIR is deprecated use EXTERNAL_BOARD_DIRS instead)
+      $(info EXTERNAL_BOARD_DIRS can contain multiple folders separated by space)
+    endif
+  endif
+endif
+
+# Needed for backward compatibility:
+ifneq (,$(BOARDSDIR))
+  EXTERNAL_BOARD_DIRS += $(BOARDSDIR)
+endif
+
+# Make all paths absolute.
+override RIOTBASE               := $(abspath $(RIOTBASE))
+override RIOTCPU                := $(abspath $(RIOTCPU))
+override RIOTMAKE               := $(abspath $(RIOTMAKE))
+override RIOTPKG                := $(abspath $(RIOTPKG))
+override RIOTTOOLS              := $(abspath $(RIOTTOOLS))
+override RIOTPROJECT            := $(abspath $(RIOTPROJECT))
+override APPDIR                 := $(abspath $(APPDIR))
+override BUILD_DIR              := $(abspath $(BUILD_DIR))
+override BINDIRBASE             := $(abspath $(BINDIRBASE))
+override BINDIR                 := $(abspath $(BINDIR))
+override PKGDIRBASE             := $(abspath $(PKGDIRBASE))
+override DLCACHE_DIR            := $(abspath $(DLCACHE_DIR))
+EXTERNAL_BOARD_DIRS             := $(foreach dir,$(EXTERNAL_BOARD_DIRS),$(abspath $(dir)))
+EXTERNAL_MODULE_DIRS            := $(foreach dir,$(EXTERNAL_MODULE_DIRS),$(abspath $(dir)))
+EXTERNAL_PKG_DIRS               := $(foreach dir,$(EXTERNAL_PKG_DIRS),$(abspath $(dir)))
+
+ifneq ($(RIOT_CI_BUILD),1)
+  ifneq ($(EXTERNAL_BOARD_DIRS),$(foreach dir,$(EXTERNAL_BOARD_DIRS),$(wildcard $(dir))))
+    $(error At least one path in EXTERNAL_BOARD_DIRS is not existing)
+  endif
+  ifneq ($(EXTERNAL_MODULE_DIRS),$(foreach dir,$(EXTERNAL_MODULE_DIRS),$(wildcard $(dir))))
+    $(error At least one path in EXTERNAL_MODULE_DIRS is not existing)
+  endif
+  ifneq ($(EXTERNAL_PKG_DIRS),$(foreach dir,$(EXTERNAL_PKG_DIRS),$(wildcard $(dir))))
+    $(error At least one path in EXTERNAL_PKG_DIRS is not existing)
```

Comment:
Can we tone this down to a warning?
I generally set `EXTERNAL_BOARD_DIRS` globally (in `~/.profile`) and not all of those paths might not exist every time e.g. because they got moved in a certain branch.

Having to keep this in sync with the repo state (and not just being able to add all possible paths) seems like an unnecessary hassle.

A warning is enough to get rid of unnecessary clutter that might accumulate.

Reply:
OK, with that use case in mind even an `$(info )` should be good enough.